### PR TITLE
fix(query) Query equality functions for tests

### PR DESCRIPTION
**Pull request by evanh**
Add query equality functions to the different query classes. Align the Logical
and Clickhouse query signature with the other query classes. Also do some mypy
cleanup, now that the logical query takes an Entity instead of a TableSource.
This isn't used much right now, but will be very useful for testing SnQL.

### CODE REVIEW

**Review by fpacifici — CHANGES_REQUESTED** [review]
Could you please add the __eq__ method to the clickhouse query as well ?
This way it would be the only query class behaving differently than the others.

**Comment by evanh** [comment]
The clickhouse query extends the ProcessableQuery class, which already has the equality. Is there another one I'm missing?

File: snuba/clickhouse/query.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 class Query(AbstractQuery[RelationalSource]):
```
Comment (fpacifici) [inline snuba/clickhouse/query.py:15]: This adds the `prewhere`  attribute. So it needs to overrided the eq method.

File: snuba/query/logical.py [hunk 2]
```diff
@@ -61,7 +61,7 @@ def __init__(
         self.__final = False
 
         super().__init__(
-            from_clause=data_source,
+            from_clause=from_clause,
```
Comment (fpacifici) [inline snuba/query/logical.py:64]: this class adds the `final` attribute so it needs to override eq.

**Review by fpacifici — CHANGES_REQUESTED** [review]
two missing fields to check.

**Review by fpacifici — APPROVED** [review]
Did you try to replace this old tech debt with your equality check ?
https://github.com/getsentry/snuba/blob/master/tests/query/parser/test_query.py#L837-L847

**Comment by evanh** [comment]
> Did you try to replace this old tech debt with your equality check ?
> https://github.com/getsentry/snuba/blob/master/tests/query/parser/test_query.py#L837-L847

Yep, I did. I forgot to delete the old checks though.

**Pull request merged**